### ISSUE TITLE
Migrate Documentation to Markdown and External Help

### ISSUE THREAD

**Issue by lipkau** [AtlassianPS/BitbucketPS#1]
## Description
In order to have a documentation that is easier to access, easier to edit and maintain, and more flexible, we shall use [PlatyPS](https://github.com/PowerShell/platyPS) for the _help_ of the function.
_Discussion on this topic: https://github.com/AtlassianPS/AtlassianPS/issues/1_

## Task
* [ ] Migrate all _Comment Based Help_ to `.md` files in `/docs`  
```posh
Import-Module MyAwesomeModule
New-MarkdownHelp -Module MyAwesomeModule -OutputFolder .\docs
```
* [ ] Edit the `.md` files (`.EXAMPLE` block can have more that 1 command in it)
* [ ] Create the `about_<module>`   
```posh
New-MarkdownAboutHelp -OutputFolder C:\Test -AboutName
```
* [ ] Make AppVeyor generate the ExternalHelp file (.xml) when building  
```posh
New-ExternalHelp -Path ".\docs" -OutputPath "out\platyPS\en-US"
```
* [ ] Remove the _Comment Based Help_ from all functions in the module

**Comment by martin9700** [comment]
I'd still recommend minimal comment based help.  SYNOPSIS and LINK would probably be best.  That way if I do Get-Help Get-Branch -Online I'd be sent to the online help page.

**Comment by lipkau** [comment]
disagree. An user should be able to get basic help (including samples) from `help <My-Command>`.
Further: the online help would not be aligned to the version the user has installed. So maybe a parameter was added, which he cannot use without updating (which will not be specified in the online help).

The online help will indeed show the latest version of the documentation of the function. but it will **not** be versioned.

An argument can be made, that generating an online .cab file with the help, which the user can get for local usage with `Update-Help`. This would allow the module to deploy fixes to the help without requiring a new release.
However, this would also require the module to be smart enough to know what help .cab file to deploy the what version of the module.

[ConfluencePS](https://github.com/AtlassianPS/ConfluencePS) has already done this. all documentation is in the `/docs` directory as `.md` files.
This allows the online help to be published as: 
https://atlassianps.org/docs/ConfluencePS/
https://atlassianps.org/docs/ConfluencePS/commands/Add-Label/

**Comment by martin9700** [comment]
Oh good! I actually prefer full comment based help, but I thought the issue was to move AWAY from that to markdown.  I assume the cbh will be pulled out at CI/CD?

**Comment by lipkau** [comment]
No. this moves away from CBH. and uses **External Help** instead...

basically, the help engine parses the comments for the CBH. this is very flawed; and used as a simple way to get help for cmdlets...
for more sophisticated help, you need to provide an external help... this can be:
* a xml file in the `/en-US` folder
* a cab file (which I have no experiance with)

pros and cons are here: https://github.com/AtlassianPS/AtlassianPS/issues/1

**Comment by martin9700** [comment]
Lol, I don't think that's the right link ;)

**Comment by lipkau** [comment]
fixed
